diff -Naur xfce4-session-4.11.0/scripts/xinitrc.in.in xfce4-session-4.11.0.tpg/scripts/xinitrc.in.in
--- xfce4-session-4.11.0/scripts/xinitrc.in.in	2014-02-22 23:01:45.000000000 +0000
+++ xfce4-session-4.11.0.tpg/scripts/xinitrc.in.in	2015-01-16 18:20:49.599145361 +0000
@@ -44,6 +44,36 @@
     xdg-user-dirs-update
 fi
 
+# Create Terminal and Thunar user preferences if they are missing
+for i in Terminal Thunar; do
+    if [ ! -d $XDG_CONFIG_HOME/$i ]; then
+	mkdir -p $XDG_CONFIG_HOME/$i
+	cp -r /etc/xdg/$i $XDG_CONFIG_HOME
+    fi ;
+done
+
+# Create panel and xfconf user preferences if they are missing
+for i in panel xfconf; do
+    if [ ! -d $XDG_CONFIG_HOME/xfce4/$i ]; then
+	mkdir -p $XDG_CONFIG_HOME/xfce4/$i
+	cp -r /etc/xdg/xfce4/$i $XDG_CONFIG_HOME/xfce4
+    fi ;
+done
+
+#(tpg) add special icons on DESKTOP
+USER_DESKTOP=`xdg-user-dir DESKTOP`
+if [ ! -e $USER_DESKTOP/om-welcome.desktop ]; then
+    cp -f /usr/share/applications/om-welcome.desktop $USER_DESKTOP 2> /dev/null
+fi
+
+if [ ! -e $USER_DESKTOP/join.desktop ]; then
+    cp -f /usr/share/applications/join.desktop $USER_DESKTOP 2> /dev/null
+fi
+
+if [ ! -e $USER_DESKTOP/donate.desktop ]; then
+    cp -f /usr/share/applications/donate.desktop $USER_DESKTOP 2> /dev/null
+fi
+
 # Modify libglade and glade environment variables so that
 # it will find the files installed by Xfce
 GLADE_CATALOG_PATH="$GLADE_CATALOG_PATH:@XFCE_GLADE_CATALOG_PATH@"
