diff -Naur xfce4-session-4.9.0/scripts/xinitrc.in.in xfce4-session-4.9.0.tpg/scripts/xinitrc.in.in
--- xfce4-session-4.9.0/scripts/xinitrc.in.in	2012-03-30 18:48:01.000000000 +0000
+++ xfce4-session-4.9.0.tpg/scripts/xinitrc.in.in	2012-04-07 22:37:14.599787190 +0000
@@ -44,6 +44,22 @@
     xdg-user-dirs-update
 fi
 
+# Create Terminal and Thunar user preferences if they are missing
+for i in Terminal Thunar; do
+    if [ ! -d $XDG_CONFIG_HOME/$i ]; then
+	mkdir -p $XDG_CONFIG_HOME/$i
+	cp -r /etc/xdg/$i $XDG_CONFIG_HOME
+    fi ;
+done
+
+# Create panel and xfconf user preferences if they are missing
+for i in panel xfconf; do
+    if [ ! -d $XDG_CONFIG_HOME/xfce4/$i ]; then
+	mkdir -p $XDG_CONFIG_HOME/xfce4/$i
+	cp -r /etc/xdg/xfce4/$i $XDG_CONFIG_HOME/xfce4
+    fi ;
+done
+
 # Modify libglade and glade environment variables so that
 # it will find the files installed by Xfce
 GLADE_CATALOG_PATH="$GLADE_CATALOG_PATH:@XFCE_GLADE_CATALOG_PATH@"
